tldw_Server_API/app/core/MCP_unified/tests/test_command_runtime_parser.py

```
@@ -0,0 +1,22 @@
+
+    assert chain.segments[0].commands[0].argv == ["cat", "notes one.txt"]
+    assert chain.segments[0].commands[1].argv == ["grep", "ERROR"]
+    assert chain.segments[1].commands[0].argv == ["write", "out.txt", "done"]
+    assert chain.links == ["&&"]
+
+
+@pytest.mark.parametrize("command", ["| grep ERROR", "cat notes.txt |", "cat &&"])
+def test_parser_rejects_malformed_sequences(command: str) -> None:
+    from tldw_Server_API.app.core.MCP_unified.command_runtime.parser import parse_command
+
+    with pytest.raises(ValueError):
```

Ruff (0.15.7)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)